packages/libs/eda/src/lib/workspace/ImportAnalysis.tsx

```diff
@@ -44,15 +55,63 @@ export function ImportAnalysis({ analysisClient, analysisId }: Props) {
           };
           history.replace(newLocation);
         },
-        (error) =>
-          setError(error instanceof Error ? error.message : String(error))
+        (error) => {
+          const enforcedErrorObject =
+            error instanceof Error ? error : new Error(String(error));
+          const errorMessageList = enforcedErrorObject.message.split('\n');
+          try {
+            setError(JSON.parse(errorMessageList[1]));
+          } catch {
+            setError(enforcedErrorObject.message);
+          }
+        }
```

Comment:
**question**: do you think we should 1) rename the argument `error` to `errorObjectOrString` or 2) add a comment in this block to explain what things `error` could be?

I read this code to myself like this:

> So this block of code is the error callback on `Task.fromPromise().chain().run()`. It accepts an error callback that's invoked with an argument we're calling `error`. What is error? Oh, I see, the type of this argument is either `Error` or something else. What's the something else? 🤔  I see we're casting it to a string which is fed into the `Error` constructor.

Maybe this only gels with me 🤷 . Below is a a summary of the two options I could think of—feel free to take one, both, or neither.

```js
        (errorObjectOrString) => {
          const error =
            error instanceof Error ? error : new Error(String(error));
          const errorMessageList = enforcedErrorObject.message.split('\n');
          try {
            setError(JSON.parse(errorMessageList[1]));
          } catch {
            setError(error.message);
          }
        }
```

```js
        (error) => {
          // `error` is either an instance of `Error` or XYZ
          const enforcedErrorObject =
            error instanceof Error ? error : new Error(String(error));
          const errorMessageList = enforcedErrorObject.message.split('\n');
          try {
            setError(JSON.parse(errorMessageList[1]));
          } catch {
            setError(enforcedErrorObject.message);
          }
        }
```